File: Q.10 Palindrome Number(leetcode)/Palindrome_Number.java
import java.util.Scanner;

public class Palindrome_Number {
    public static boolean isPalindrome(int x) {
        int duplicate_x = x;
        long rev = 0;
        if (x < 0) {
            return false;
        }
        while (x != 0) {
            rev = rev * 10 + (x % 10);
            x /= 10;
        }
        return (rev == duplicate_x);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        System.out.println(isPalindrome(n));
        sc.close();
    }
}

File: Q.10 Palindrome Number(leetcode)/Palindrome_Number.py
class Solution(object):
    def isPalindrome(self, x):
        duplicate=x
        rev=0
        if x<0:
            return False
        while(duplicate!=0):
            rev=rev*10+(duplicate%10)
            duplicate//=10
        if rev==x:
            return True
        else:
            return False
        
solution=Solution()
n=int(input())
print(solution.isPalindrome(n))